sql/core/src/main/scala/org/apache/spark/sql/DataFrameImpl.scala

```diff
@@ -88,12 +88,24 @@ private[sql] class DataFrameImpl protected[sql](
     }
   }
 
-  protected[sql] def numericColumns: Seq[Expression] = {
-    schema.fields.filter(_.dataType.isInstanceOf[NumericType]).map { n =>
-      queryExecution.analyzed.resolve(n.name, sqlContext.analyzer.resolver).get
+  protected[sql] def numericColumns(colNames: String*): Seq[Expression] = {
+    val allNumbericCols = schema.fields.filter(_.dataType.isInstanceOf[NumericType]).map(_.name)
+    val diff = colNames.diff(allNumbericCols)
+    if (diff.nonEmpty) {
```

Comment:
i think an easier way is to just traverse colNames, and throw an exception if it is a not a numeric column. Really no need to be so complicated to get the diff ...